### PR TITLE
Links

### LINKED ISSUE #35: Make links in messages clickable in the Swing GUI
When in a chat room and a message with a link occurs,
Then allow the user to click and open the link.

### PR DESCRIPTION
**Pull request by Limvot**
Fixes #35! Works under Wayland now by removing a bad env var that gets set, see the comments in the code.

Depends/based on #37

### CODE REVIEW

**Review by lotusronin — COMMENTED** [review]
Looking good to me, though does this belong in the UI layer? Feels like most every UI will want to launch a program or app  from some command or input data which makes me think it maybe should eventually go into the shared code with platform specific backends?

File: serif_swing/src/main/kotlin/serif_swing/App.kt [hunk 1]
```diff
@@ -208,13 +266,96 @@ class SwingChatRoom(val transition: (MatrixState, Boolean) -> Unit, val panel: J
...
+                    // It's not in the examples I found online
+                    // My best guess is that because of the layout, it
+                    // won't smash it smaller than preferred size, but it will
+                    // stretch it to fit the larger size?
+                    message.setPreferredSize(Dimension(0, 0))
+
+                    var current_idx = 0
+                    val simpleAttrs = SimpleAttributeSet()
+                    for (url_match in URL_REGEX.findAll(msg.message)) {
+                        println("${url_match.value} is a url!!")
```
Comment (lotusronin) [inline serif_swing/src/main/kotlin/serif_swing/App.kt]: We should remove this print given the logging that occurs when attempting to open the link

**Review by lotusronin — CHANGES_REQUESTED** [review]

**Comment by Limvot** [comment]
> Looking good to me, though does this belong in the UI layer? Feels like most every UI will want to launch a program or app from some command or input data which makes me think it maybe should eventually go into the shared code with platform specific backends?

Maybe, that's an interesting point - I was wondering if the URL detection should go in the SharedUI layer too, and then we pass up which parts of the message are URLs? On the other hand, clicking on a link doesn't do anything from the SharedUI state, so passing through it to get to the platform implementation seems unnecessary, unless we have multiple GUIs per platform and want to consolidate hacks like this... I'm really not sure!

File: serif_swing/src/main/kotlin/serif_swing/App.kt (hunk 1 above)
Comment (Limvot) [reply serif_swing/src/main/kotlin/serif_swing/App.kt]: Fair, done!

**Review by lotusronin — APPROVED** [review]
:+1:  lgtm

**Pull request merged**